include/FastBVH/Vector3.h

```diff
@@ -64,6 +67,11 @@ Float dot(const Vector3<Float>& a, const Vector3<Float>& b) noexcept {
   return (a.x * b.x) + (a.y * b.y) + (a.z * b.z);
 }
 
+template <typename Float>
```

Comment:
This is just what I usually refer to as component wise multiplication. The Hadamard product is just another name for it.